main.go

```diff
@@ -79,6 +80,7 @@ func (s *SuperAgent) ClearSuperAgent() {
 	s.TargetType = "json"
 	s.Cookies = make([]*http.Cookie, 0)
 	s.Errors = nil
+	s.BasicAuth = nil
```

Comment:
My mistake, you are right. It is not related to New().
But it is actually an issue if we cannot clear BasicAuth's content when we need to.

Also, it depends on how you want to use it. For example, if you are reusing gorequest agent for some other routing that not need auth, how would you do it?
```
request := gorequest.New().SetBasicAuth("me","mypass")
request.Get("me.com/auth").End()
request.Get("me.com/noauth").End()
```
Above would not work. With your old implementation, "me.com/noauth" will also get BasicAuth content.
But if you use your new implementation which you add nil in ClearSuperAgent(). This will work only if we set it for every routing as follows:
```
request := gorequest.New()
request.Get("me.com/auth").SetBasicAuth("me","mypass").End()
request.Get("me.com/noauth").End()
```